cypress/integration/dashboard/dashboard_spec.js

```diff
@@ -1,59 +1,145 @@
     return slug;
   });
 }
 
-function archiveCurrentDashboard() {
+function editDashboard() {
   cy.getByTestId('DashboardMoreMenu')
     .click()
     .within(() => {
       cy.get('li')
-        .contains('Archive')
+        .contains('Edit')
         .click();
     });
+}
+
+function addTextbox(text) {
+  cy.server();
+  cy.route('POST', 'api/widgets').as('NewWidget');
+
+  editDashboard();
+
+  cy.contains('a', 'Add Textbox').click();
+  cy.get('.add-textbox').within(() => {
+    cy.get('textarea').type(text);
+  });
+  cy.contains('button', 'Add to Dashboard').click();
+  cy.get('.add-textbox').should('not.exist');
+  cy.contains('button', 'Apply Changes').click();
 
-  cy.get('.btn-warning')
-    .contains('Archive')
-    .click();
-  cy.get('.label-tag-archived').should('exist');
+  return cy.wait('@NewWidget').then((xhr) => {
+    const id = Cypress._.get(xhr, 'response.body.id');
+    assert.isDefined(id, 'Widget api call returns widget id');
+    return cy.getByTestId(`WidgetId${id}`);
+  });
 }
 
 describe('Dashboard', () => {
   beforeEach(() => {
     cy.login();
   });
 
-  it('creates a new dashboard and archives it', () => {
-    createNewDashboard('Foo Bar').then((slug) => {
+  it('creates new dashboard', () => {
+    cy.visit('/dashboards');
+    cy.getByTestId('CreateButton').click();
+    cy.get('li[role="menuitem"]').contains('Dashboard').click();
+
+    cy.server();
+    cy.route('POST', 'api/dashboards').as('NewDashboard');
+
+    cy.getByTestId('EditDashboardDialog').within(() => {
+      cy.getByTestId('DashboardSaveButton').should('be.disabled');
+      cy.get('input').type('Foo Bar');
+      cy.getByTestId('DashboardSaveButton').click();
+    });
+
+    cy.wait('@NewDashboard').then((xhr) => {
+      const slug = Cypress._.get(xhr, 'response.body.slug');
+      assert.isDefined(slug, 'Dashboard api call returns slug');
+
       cy.visit('/dashboards');
       cy.getByTestId('DashboardLayoutContent').within(() => {
-        cy.getByTestId(slug).should('exist').click();
+        cy.getByTestId(slug).should('exist');
       });
+    });
+  });
+
+  it('archives dashboard', function () {
+    createNewDashboardByAPI('Foo Bar').then((slug) => {
+      cy.visit(`/dashboard/${slug}`);
 
-      archiveCurrentDashboard();
+      cy.getByTestId('DashboardMoreMenu')
+        .click()
+        .within(() => {
+          cy.get('li')
+            .contains('Archive')
+            .click();
+        });
+
+      cy.get('.btn-warning')
+        .contains('Archive')
+        .click();
+      cy.get('.label-tag-archived').should('exist');
 
       cy.visit('/dashboards');
       cy.getByTestId('DashboardLayoutContent').within(() => {
         cy.getByTestId(slug).should('not.exist');
       });
     });
   });
+
+
+  describe('Textbox', () => {
+    before(function () {
+      cy.login();
+      createNewDashboardByAPI('Foo Bar')
+        .then(slug => `/dashboard/${slug}`)
+        .as('DashboardUrl');
```

Comment:
Why Upper? it looks weird when you do `this.DashboardUrl` :joy:

Reply:
For some reason I thought it's the cypress convention.. I'll change back.